### PR TITLE
Correct typos, simplify code

### PR DESCRIPTION
**Pull request by software-made-easy**

### CODE REVIEW

File: markdownhighlighter.cpp [hunk 1]
```diff
@@ -1234,7 +1234,7 @@ int MarkdownHighlighter::highlightNumericLiterals(const QString &text, int i) {
     bool isPostfixAllowed = false;
     if (i == text.length()) {
         // cant have e at the end
-        if (isCurrentHex || (!isCurrentHex && text.at(i - 1) != QChar('e'))) {
+        if (isCurrentHex || text.at(i - 1) != QChar('e')) {
```
Comment (software-made-easy) [inline markdownhighlighter.cpp:1237]: Is the same only simplified

File: markdownhighlighter.cpp [hunk 2]
```diff
@@ -1420,8 +1420,7 @@ void MarkdownHighlighter::ymlHighlighter(const QString &text) {
 
         // underlined links
         if (text.at(i) == QChar('h')) {
-            if (MH_SUBSTR(i, 5) == QLatin1String("https") ||
-                MH_SUBSTR(i, 4) == QLatin1String("http")) {
+            if (MH_SUBSTR(i, 4) == QLatin1String("http")) {
```
Comment (software-made-easy) [inline markdownhighlighter.cpp:1423]: We don't need to look at http and https, http is enough

**Comment by pbek** [comment]
@Waqar144?

File: markdownhighlighter.cpp [hunk 3]
```diff
@@ -248,7 +248,7 @@ void MarkdownHighlighter::initHighlightingRules() {
 
     // highlight inline comments
     {
-        // highlight comments for Rmarkdown for academic papers
+        // highlight comments for Markdown for academic papers
```
Comment (Waqar144) [inline markdownhighlighter.cpp:251]: I think RMarkdown is correct here?

File: markdownhighlighter.cpp [hunk 4]
```diff
@@ -1749,7 +1748,7 @@ void MarkdownHighlighter::highlightCheckbox(const QString &text, int curPos)
     const bool hasXorSpace = midChar == QLatin1Char(' ') || midChar == QLatin1Char('x') || midChar == QLatin1Char('X');
     const bool hasDash = midChar == QLatin1Char('-');
 
-    if (hasOpeningBracket && hasClosingBracked && (hasXorSpace | hasDash)) {
+    if (hasOpeningBracket && hasClosingBracked && (hasXorSpace || hasDash)) {
```
Comment (Waqar144) [inline markdownhighlighter.cpp:1751]: Spelling of hasClosingBracked can also be fixed if you want to

**Review by Waqar144 — APPROVED** [review]

File: markdownhighlighter.cpp (hunk 3 above)
Comment (pbek) [reply markdownhighlighter.cpp]: I think R Markdown

File: markdownhighlighter.cpp (hunk 4 above)
Comment (pbek) [reply markdownhighlighter.cpp]: Good idea

File: markdownhighlighter.cpp [hunk 5]
```diff
@@ -73,7 +73,7 @@ void MarkdownHighlighter::timerTick() {
     // emit a signal every second if there was some highlighting done
     if (_highlightingFinished) {
         _highlightingFinished = false;
-        emit highlightingFinished();
+        Q_EMIT highlightingFinished();
```
Comment (pbek) [inline markdownhighlighter.cpp:76]: @Waqar144, are you ok with the Q_EMITs?

Comment (Waqar144) [reply markdownhighlighter.cpp:76]: Yes, it's better. However I would personally suggest you to convert them all in one go, maybe in a separate commit/PR. emit, signals, slots etc

**Pull request merged**